server/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -138,7 +159,9 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
                     ShardPath shardPath = null;
                     try {
                         IndexSettings indexSettings = new IndexSettings(metaData, settings);
-                        shardPath = ShardPath.loadShardPath(logger, nodeEnv, shardId, indexSettings);
+                        try (ShardLock ignored = nodeEnv.shardLock(shardId, TimeUnit.SECONDS.toMillis(5))) {
```

Comment:
did you double check the implications of a `ShardLockObtainFailedException`

Reply:
We obtain the same shard lock a few lines down, in `Store.tryOpenIndex(...)`, unless `ShardPath.loadShardPath` returns `null` in which case we throw a different exception.